src/main/java/com/redhat/rhjmc/containerjfr/tui/tcp/SocketClientReaderWriter.java

```diff
@@ -53,6 +55,20 @@
         listenerThread.start();
     }
 
+    // Testing-only constructor
+    SocketClientReaderWriter(Logger logger, Semaphore semaphore, Socket s, Scanner scanner, OutputStreamWriter writer) {
+        this.logger = logger;
+        this.semaphore = semaphore;
+        this.scanner = scanner;
+        this.writer = writer;
+        this.listenerThread = null;
+        this.ss = null;
+        this.s = s;
```

Comment:
Since you're already here refactoring, could you please rename `s` and `ss` more appropriately?

Reply:
Yes. This will be fixed in the next commit.